lib/daisy_ui_components/steps.ex

```diff
@@ -0,0 +1,92 @@
+defmodule DaisyUIComponents.Steps do
+  @moduledoc """
+  Steps component used to display a sequence of steps in a process.
+
+  ## Basic Example:
+      <.steps>
+        <.step class="step-primary">Register</.step>
+        <.step class="step-primary">Choose plan</.step>
+        <.step>Purchase</.step>
+        <.step>Receive Product</.step>
+      </.steps>
+
+  ## Renders:
+      <ul class="steps">
+        <li class="step step-primary">Register</li>
+        <li class="step step-primary">Choose plan</li>
+        <li class="step">Purchase</li>
+        <li class="step">Receive Product</li>
+      </ul>
+  """
+
+  use DaisyUIComponents, :component
+
+  attr :class, :any, default: nil
+  attr :vertical, :boolean, default: false
+  attr :horizontal, :boolean, default: false
+  attr :rest, :global
+
+  slot :inner_block
+
+  def steps(assigns) do
+    assigns =
+      assign(assigns, :class, steps_classes(assigns))
+
+    ~H"""
+    <ul class={@class} {@rest}>
+      {render_slot(@inner_block)}
+    </ul>
+    """
+  end
+
+  attr :class, :any, default: nil
+  attr :content, :string, default: nil
+  attr :neutral, :boolean, default: false
+  attr :primary, :boolean, default: false
+  attr :secondary, :boolean, default: false
+  attr :accent, :boolean, default: false
+  attr :info, :boolean, default: false
+  attr :success, :boolean, default: false
+  attr :warning, :boolean, default: false
+  attr :error, :boolean, default: false
+  attr :rest, :global
+
```

Comment:
regarding the icons, adding an icon attribute could be good too:

```elixir
attr :icon, :string, default: nil

<.step icon="😕" />
```
which would render the `step-icon` way. I need to see also if this can play out with the heroicons, probably not but if we add this simple example/case would be enough - having icon attribute and you just put the emoji